src/payment/strategies/klarnav2/klarnav2-payment-strategy.ts

```diff
@@ -0,0 +1,186 @@
+                );
+
+                return this._loadWidget(options);
+            })
+            .then(() => this._store.getState());
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        if (this._unsubscribe) {
+            this._unsubscribe();
+        }
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        if (!payload.payment) {
+            throw new InvalidArgumentError('Unable to proceed because "payload.payment" argument is not provided.');
+        }
+
+        const { payment: { paymentData, ...paymentPayload } } = payload;
+
+        return this._authorize(paymentPayload.methodId)
+            .then(({ authorization_token: authorizationToken }) => this._store.dispatch(
+                this._remoteCheckoutActionCreator.initializePayment(paymentPayload.methodId, { authorizationToken })
+            ))
+            .then(() => this._store.dispatch(
+                this._orderActionCreator.submitOrder({
+                    ...payload,
+                    payment: paymentPayload,
+                    // Note: API currently doesn't support using Store Credit with Klarna.
+                    // To prevent deducting customer's store credit, set it as false.
+                    useStoreCredit: false,
+                }, options)
+            ));
+    }
+
+    finalize(): Promise<InternalCheckoutSelectors> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    private _loadWidget(options: PaymentInitializeOptions): Promise<KlarnaLoadResponse> {
+        if (!options.klarnav2) {
+            throw new InvalidArgumentError('Unable to load widget because "options.klarna" argument is not provided.');
+        }
+
+        const {methodId, klarnav2: {container, payment_method_category, onLoad}} = options;
+        const state = this._store.getState();
+
+        return new Promise<KlarnaLoadResponse>(resolve => {
+            const paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+            if (!paymentMethod) {
+                throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+            }
+
+            if (!this._klarnaPayments || !paymentMethod.clientToken) {
+                throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+            }
+            this._klarnaPayments.init({client_token: paymentMethod.clientToken});
+            this._klarnaPayments.load({container, payment_method_category}, response => {
+                if (onLoad) {
+                    onLoad(response);
+                }
+                resolve(response);
+            });
+        });
+    }
+
+    private _getUpdateSessionData(billingAddress: BillingAddress, shippingAddress?: Address): KlarnaUpdateSessionParams {
+        if (!includes(this._supportedEUCountries, billingAddress.countryCode)) {
```

Comment:
You should send this directly from BCApp

```JavaScript
        const { methodId } = options;

        this._paymentMethod = this._store.getState().paymentMethods.getPaymentMethod(methodId);

        if (!this._paymentMethod || !this._paymentMethod.initializationData) {
            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
        }

        const _supportedCountries = { !this._paymentMethod.initializationData };

```

Reply:
As far as I know checkout team does not want us to use `initializationData` field unless it is really necessary due this field is an `Any` type